research/src/pages/popup/popup.proposal.mdx

```diff
@@ -0,0 +1,418 @@
+  special case.
+</p>
+
+When a `popup` element is shown and a light dismiss interaction occurs:
+
+1. If a keypress event for the `ESCAPE` key was fired, the user agent must set the _subject_ to
+   the `popup` element and run [hiding a `popup` element steps](#hiding-popup-steps).
+2. If a focus change occurs, the user agent must set the _start node_ to the element which received
+   the focus event and run [Hiding currently-shown `popup` elements steps](#hiding-current-popups).
+3. Otherwise, the user agent must set the _start node_ to null and run [Hiding currently-shown
+   `popup` elements steps](#hiding-current-popups).
+
+<h3 id="hiding-current-popups">
+  Hiding currently-shown <code>popup</code> elements steps
+</h3>
+
+Starting at the top of the [popup stack](#popup-stack):
+
+1. Let _subject_ be the current `popup` element in the [popup stack](#popup-stack).
+2. If _candidate subject_ is null, skip to step 6.
+3. Otherwise, starting with the _candidate subject_, walk the
+   [flat tree](https://drafts.csswg.org/css-scoping/#flattening) to determine whether the _subject_ is
+   an ancestor to the _candidate subject_. If the _subject_ is an ancestor, then return.
+4. Otherwise, if the _invoker_ is not null, starting with the _invoker_, walk the
+   [flat tree](https://drafts.csswg.org/css-scoping/#flattening) to determine whether the _subject_ is
+   an ancestor to the _invoker_. If the _subject_ is an ancestor, then return.
+5. Otherwise, if the _anchor element_ is not null, starting with the _anchor element_, walk the
+   [flat tree](https://drafts.csswg.org/css-scoping/#flattening) to determine whether the _subject_ is
+   an ancestor to the _anchor element_. If the _subject_ is an ancestor, then return.
+6. If _start node_ is not null, starting with the _start node_, walk the
+   [flat tree](https://drafts.csswg.org/css-scoping/#flattening) to determine whether the _subject_ is
+   the _start node_ or an ancestor to the _start node_. If either condition is met, then return.
+7. Otherwise, for this _subject_, run the
+   [hiding a `popup` element steps](#hiding-popup-steps). Repeat for the next `popup` element in
+   the [popup stack](#popup-stack).
+
+<h3 id="hiding-popup-steps">
+  Hiding a <code>popup</code> element steps
+</h3>
+
+The hiding steps for a `popup` element _subject_ are as follows:
+
+1. Remove the _subject_ from the
+   [node document](https://dom.spec.whatwg.org/#concept-node-document)'s
+   [top layer](https://fullscreen.spec.whatwg.org/#top-layer).
+2. Set the `open` IDL attribute on the _subject_ to `false`.
+3. If applicable, empty the _subject_'s _invoker_.
+4. [Queue an element task](https://html.spec.whatwg.org/multipage/webappapis.html#queue-an-element-task)
+   on the [user interaction task source](https://html.spec.whatwg.org/multipage/webappapis.html#user-interaction-task-source)
+   given the _subject_ element to [fire a non-cancelable event](https://dom.spec.whatwg.org/#concept-event-fire)
```

Comment:
```suggestion
   given the _subject_ element to [fire an event](https://dom.spec.whatwg.org/#concept-event-fire)
```

Reply:
Added some changes to ccd9b75036cb67e177ad11b2bf72b0b9c1ed0afd to address this feedback. Please let me know if the changes resolve the issue or if you have any further comments; feel free to hold until I've resolved all open feedback and have formally requested a new review.